### ISSUE TITLE
Bug in the webmonitor.py example that causes just one iteration and then stops

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11313]
|<img alt="lstep's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| lstep reported|
|-|-|
|Trac ID|trac#308|
|Type|defect|
|Created|2003-10-02 00:31:14Z|

<details><summary>Searchable metadata</summary>

```
trac-id__308 308
type__defect defect
reporter__lstep lstep
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1065054674000000 1065054674000000
changetime__1468712242265269 1468712242265269
version__ 
owner__ 
cc__lstep cc__moshez
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="lstep's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|lstep commented|
|-|-|

```
#!html
<pre>
The webmonitor.py example, when run, connects just once to the remote web 
server, and then do nothing. The problem is that in the example there is no 
reportNoChange() method in the ChangeNotified class. So when the ChangeChecker 
is called and that there is no change on the page (most usual :), it tries to 
call self.notified.reportNoChange() which doesn't exist ! So it "crashes" 
silently, and you think monitor is working... 
To correct this, just add:: 
 
  def reportNoChange(self): 
          pass 
in the ChangeNotified class of the webmonitor.py example file. 
 
To be more clean, I suggest to add a test for the presence of the methods 
reportNoChange() and reportChange() in t.w.monitor module.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Checking in twisted/web/monitor.py;
/cvs/Twisted/twisted/web/monitor.py,v  &lt;--  monitor.py
new revision: 1.7; previous revision: 1.6
done
Fixed [By enhancing BaseChange]
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset 778eda4f6feb8f0cfebc9489488b3ac3d1f672b9
```
#!CommitTicketReference repository="" revision="778eda4f6feb8f0cfebc9489488b3ac3d1f672b9"
Merge pull request #308 from twisted/8573-scop-files

Author: scop
Reviewer: rodrigc
Fixes: #8573

More file close related fixes
```
